typescriptRoutes/jsonPlaceholder.ts

```diff
@@ -0,0 +1,105 @@
+import * as Router from "koa-router";
+import { KoaContext } from "../types";
+import { routeHelper } from "./routerHandler";
+import { IHttpClient } from "../interface";
+import { HttpClient } from "../middleware/httpClient";
+import { methods } from "../types";
+
+interface IJsonPlaceHolder {
+    handleGetRequest: () => {};
+    handlePostRequest: (ctx: KoaContext) => {};
+    handlePutRequest?: (ctx: KoaContext) => {};
+    handleDeleteRequest?: (ctx: KoaContext) => {};
+}
+
+export class JsonPlaceHolder implements IJsonPlaceHolder {
+    public http: IHttpClient;
+    constructor(httpClient: IHttpClient) {
+        this.http = httpClient;
+    }
+
+    handleGetRequest = async () => {
+        const res = await this.http.get(
+            "https://jsonplaceholder.typicode.com/posts",
+            {}
+        );
+        return res.data;
+    };
+    handlePostRequest = async (ctx: KoaContext) => {
```

Comment:
you need to override the KoaContext and write something lik 
ctx = KoaContext & {
   body: <your_type>
   params: <your_type>
   ...
}